python/tutorials/11-programmatic-dependent-launch.py

```diff
@@ -0,0 +1,112 @@
+"""
+Programmatic Dependent Launch
+=====================
+This script demonstrates the use of programmatic dependent launch ontop of the vector-add example using Triton.
+
+.. code-block:: bash
+    python 11-programmatic-dependent-launch.py
+"""
+
+import torch
+import triton
+import triton.language as tl
+
+
+def is_cuda():
+    return triton.runtime.driver.active.get_current_target().backend == "cuda"
+
+
+def supports_pdl():
+    return is_cuda() and torch.cuda.get_device_capability()[0] >= 9
+
+
+# In this example
+@triton.jit
+def add_kernel(x_ptr,  #
+               y_ptr,  #
+               output_ptr,  #
+               n_elements,  #
+               BLOCK_SIZE: tl.constexpr,  #
+               USE_GDC: tl.constexpr,  #
+               ):
+    pid = tl.program_id(axis=0)
+    block_start = pid * BLOCK_SIZE
+    offsets = block_start + tl.arange(0, BLOCK_SIZE)
+    mask = offsets < n_elements
+    if USE_GDC:
+        # GDC wait is used to wait for the prior kernel to complete before continuing.
+        # If we utilize Programmatic Dependent Launch, we must wait on the prior kernel
+        # to complete in case a or b are written to by the prior kernel.
+        # This is done to prevent races.
+        tl.extra.cuda.gdc_wait()
+
+    x = tl.load(x_ptr + offsets, mask=mask)
+    y = tl.load(y_ptr + offsets, mask=mask)
+    if USE_GDC:
+        # GDC launch dependents is used to launch dependent kernels.
+        # Once GDC launch it is possible for the next kernel to begin if
+        # there are enough resources.
```

Comment:
```suggestion
        # GDC launch dependents is used to launch dependent kernels.
        # Once GDC launch has been issued at least once on all active hardware SMs, 
        # it is possible for the next kernel to begin if there are enough resources.
        # Note this provides no additional memory-ordering guaruntees, unlike `gdc_wait`.
```

Reply:
Is this correct? There are no suggestions of memory ordering in the docs.
https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-griddepcontrol